sys-utils/cartesi-init/cartesi-init

```diff
@@ -4,11 +4,11 @@
 busybox stty -onlcr
 
 # mount
-busybox mkdir -p /dev/pts /dev/shm
-busybox mount -o nosuid,nodev,noexec          -t proc proc /proc
-busybox mount -o nosuid,noexec,mode=620,gid=5 -t devpts devpts /dev/pts
-busybox mount -o nosuid,nodev,mode=1777       -t tmpfs tmpfs /dev/shm
-busybox mount -o nosuid,nodev,noexec          -t sysfs sys /sys
+[ -d /dev ] && busybox mkdir -p /dev/pts /dev/shm
```

Comment:
I'm not sure about this. We'll be progressing the boot on a broken runtime.
A bunch of commands require these directories to exist and to be mounted.

Have you faced a use case for something like this? can you go into more detail?

Reply:
The goal was to allow using the tools in distroless filesystem trees, where the filesystem is almost empty (just tools + GLIBC + busybox files).

Needing `/proc` and `/dev` is fair, I reverted these changes.
But I added the creation of `/proc` now in case it does not exists, and to mount `/sys` only if present.
I think `/sys` is not mandatory.

Reply:
> I think /sys is not mandatory.

I did a quick search but couldn't out if it is optional or mandatory.

> The goal was to allow using the tools in distroless filesystem trees, where the filesystem is almost empty (just tools + GLIBC + busybox files).

What about creating them when needed, will this work for you?
```
# mount
busybox mkdir -p /proc /dev/pts /dev/shm /sys /tmp /run
busybox mount -o nosuid,nodev,noexec          -t proc proc /proc
busybox mount -o nosuid,noexec,mode=620,gid=5 -t devpts devpts /dev/pts
busybox mount -o nosuid,nodev,mode=1777       -t tmpfs tmpfs /dev/shm
busybox mount -o nosuid,nodev,noexec -t sysfs sys /sys
busybox mount -o nosuid,nodev,mode=1777 -t tmpfs tmpfs /tmp
busybox mount -o nosuid,nodev,mode=0755 -t tmpfs tmpfs /run
```

Reply:
Yep, will do.

Reply:
Done that, also added creation and mount of `/dev` in case it does not exist. I could make a distroless rootfs of tools including just `rollup-http-server` with the following script:

```
set -e
# Create file system tree
mkdir -p rootfs/bin rootfs/sbin rootfs/usr/lib rootfs/root rootfs/etc
echo "root:x:0:0:root:/root:/bin/sh" > rootfs/etc/passwd
ln -sf busybox rootfs/bin/sh
# Copy tools
wget -O rootfs-tools-v0.14.1.ext2 https://github.com/cartesi/machine-emulator-tools/releases/download/v0.14.1/rootfs-tools-v0.14.1.ext2
e2cp -p rootfs-tools-v0.14.1.ext2:/lib/riscv64-linux-gnu/ld-linux-riscv64-lp64d.so.1 rootfs/lib/
e2cp -p rootfs-tools-v0.14.1.ext2:/lib/riscv64-linux-gnu/libgcc_s.so.1 rootfs/lib/
e2cp -p rootfs-tools-v0.14.1.ext2:/lib/riscv64-linux-gnu/libm.so.6 rootfs/lib/
e2cp -p rootfs-tools-v0.14.1.ext2:/lib/riscv64-linux-gnu/libc.so.6 rootfs/lib/
e2cp -p rootfs-tools-v0.14.1.ext2:/usr/bin/rollup-http-server rootfs/bin/
e2cp -p rootfs-tools-v0.14.1.ext2:/usr/bin/busybox rootfs/bin/
wget -O rootfs/bin/cartesi-init https://raw.githubusercontent.com/cartesi/machine-emulator-tools/f8cc3aafc50e73558b575f5262e957e1ab6ea68e/sys-utils/cartesi-init/cartesi-init
# Generate rootfs
xgenext2fs -fzB 4096 -i 4096 --readjustment +1024 --squash-uids 0 --root rootfs rootfs.ext2
# Test it
cartesi-machine --flash-drive=label:root,filename:rootfs.ext2 rollup-http-server --help
```

Output:
```
$ cartesi-machine --flash-drive=label:root,filename:rootfs.ext2 rollup-http-server --help

         .
        / \
      /    \
\---/---\  /----\
 \       X       \
  \----/  \---/---\
       \    / CARTESI
        \ /   MACHINE
         '

Usage: rollup-http-server [options] <command> [args]

Where command and args start the DApp.

Options:
    -h, --help          show this help message and exit
        --address       Address to listen (default: 127.0.0.1:5004)
        --dapp          Dapp address (default: 127.0.0.1:5003)
        --verbose       print more info about application execution

Halted
Cycles: 53695017
```

Reply:
Cool!